### PR TITLE
refactor(test): tests/floodsub - addition of test_factory.rs

### PR DESCRIPTION
**Pull request by Ivansete**
Moving implementation of `new_test_topic()` and `new_test_node()` into a separate file: test_factory.rs.

### CODE REVIEW

**Review by LNSD — APPROVED** [review]
In general, LGTM ✅ 

Just a tiny nitpicking comment:

> As in the [pubsub-core/tests](https://github.com/LNSD/rust-libp2p-pubsub/tree/main/pubsub-core/tests) directory, the `pubsub_testlib` module is in the root of the directory; I'd rather have a `floodsub_testlib` module in the `/floodsub/tests` directory instead of within a test suite directory.

**Comment by Ivansete** [comment]
@LNSD - I've applied a different approach. Let me know what do you think about that one :)

File: testlib/Cargo.toml
Comment (LNSD) [inline testlib/Cargo.toml:1]: Adding a sub-crate under the tests directory is too much. Additionally, this "local testlib" will clash with the global testlib. 

The lightest option is adding a `floodsub_testlib` module in the `floodsub/tests` directory. And "import it" in each module that uses it (as it suggests the rust test organization documentation: [here](https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests)).

In the future, we may move things to the global `testlib`, but for now, let's keep things local to the crate without additional sub-crates.

**Review by LNSD — CHANGES_REQUESTED** [review]
Please, check my comments.

**Review by LNSD — APPROVED** [review]

**Pull request merged**